zerver/lib/streams.py

```diff
@@ -185,6 +187,10 @@ def check_stream_name(stream_name: str) -> None:
                 _("Stream name '{}' contains NULL (0x00) characters.").format(stream_name)
```

Comment:
This is covered by `Cc`, below.  It's a more specific error message, but I don't know that it merits still having its own check.

Reply:
I think we can remove this line and just keep the existing test (with the new more generic string).  

We added this because it appeared in the wild, but there's no reason to have a duplicated check for it now that we're doing something better.

Reply:
Should we mention the category or character in the error message

Reply:
Putting the character in the error message is nice, in that it's specific about what needs to be changed -- but bear in mind that we'd need to come up with some way to escape them.  Otherwise we'd say:

```
"Invalid character 
 in stream name!"
```

...which might be confusing.  And while [it's not prohibited to send back a non-character](https://www.unicode.org/faq/private_use.html#nonchar7) if that's what we got sent, it'd be nice to send it as `\xFFFF` and not `￿`.

So, if we can do it carefully, it's a fine idea.  Note that it will only show the _first_ offending character, not all of them.

Reply:
@alexmv @timabbott  I think we stringify `Cc` characters but as you mentioned non-characters aren't sent back, and besides we may also face problems with surrogate characters too if that case ever happens. 
I think `Invalid unicode characters in stream name!` would be a better error message. What are your views about this?

Reply:
> I think we stringify `Cc` characters

I'm not sure what you mean by "stringify" here.  We don't currently send back the offending character in any form, so I don't think there's anything to say about what we currently do with `Cc` characters.

> as you mentioned non-characters aren't sent back

We do _currently_ (before this PR) faithfully use any instances of `\xFFFF`, like any other character.  I was talking about if it was going to be valid to send the characters back in an error message.  The link I pointed to said doing so was acceptible.

> and besides we may also face problems with surrogate characters too if that case ever happens

I strongly suspect that we won't actually be able to get half of a surrogate pair this far down the stack, since that requires UTF-16 encoding.  One point is that UTF-8 can't actually encode codepoints that are surrogate pairs, so I don't know how we'd get them here.  But we certainly wouldn't be able to send them back in the error message, yes.

> I think `Invalid unicode characters in stream name!` would be a better error message.

I don't think we should say they're "unicode characters" -- they're just characters.  And I think I made my opinion clear in my previous message -- if we can carefully tell the user the offending character, that would be useful.

I guess one simpler alternative would be to tell them which character it is by _position_, instead of sending it back.  So have the error be `Invalid character in stream name, at position {x}!`